app/src/main/java/com/android/feedme/model/viewmodel/ProfileViewModel.kt

```diff
@@ -24,29 +24,42 @@ class ProfileViewModel : ViewModel() {
   var currentUserId: String? = null
   private val _currentUserProfile = MutableStateFlow<Profile?>(null)
   val currentUserProfile: StateFlow<Profile?> = _currentUserProfile
-  private val _currentUserFollowers = MutableStateFlow<List<Profile>>(listOf(Profile()))
-  private val _currentUserFollowing = MutableStateFlow<List<Profile>>(listOf(Profile()))
+  private val _currentUserFollowers = MutableStateFlow<List<Profile>>(listOf())
+  private val _currentUserFollowing = MutableStateFlow<List<Profile>>(listOf())
   val currentUserFollowers: StateFlow<List<Profile>> = _currentUserFollowers
   val currentUserFollowing: StateFlow<List<Profile>> = _currentUserFollowing
 
   // Viewing User
   var viewingUserId: String? = null
   private val _viewingUserProfile = MutableStateFlow<Profile?>(null)
-  private val _viewingUserFollowing = MutableStateFlow<List<Profile>>(listOf(Profile()))
-  private val _viewingUserFollowers = MutableStateFlow<List<Profile>>(listOf(Profile()))
+  private val _viewingUserFollowing = MutableStateFlow<List<Profile>>(listOf())
+  private val _viewingUserFollowers = MutableStateFlow<List<Profile>>(listOf())
   val viewingUserProfile: StateFlow<Profile?> = _viewingUserProfile
   val viewingUserFollowing: StateFlow<List<Profile>> = _viewingUserFollowing
   val viewingUserFollowers: StateFlow<List<Profile>> = _viewingUserFollowers
 
+  // Listen to FirebaseAuth state changes, to fetch the profile of the current user
+  private val authListener =
+      FirebaseAuth.AuthStateListener { firebaseAuth ->
+        val user = firebaseAuth.currentUser
+        user?.uid?.let {
+          currentUserId = it
+          fetchCurrentUserProfile()
+        }
+      }
+
+  // Error message
+  private val _errorMessages = MutableStateFlow<String?>(null)
```

Comment:
No, this could be used to explain to the User the error, for example, it could be that your are offline and cannot to Firestore anymore. This variable allow to communicate those infos to the user